lib/resolve.go

```diff
@@ -51,107 +46,62 @@ func (inst *Instance) ParseAndResolveRefWithWorkingDir(ctx context.Context, refS
 }
 
 // ResolveReference finds the identifier & HEAD path for a dataset reference.
-// the source parameter determines which subsystems of Qri to use when
-func (inst *Instance) ResolveReference(ctx context.Context, ref *dsref.Ref, source string) (string, error) {
+// the mode parameter determines which subsystems of Qri to use when resolving
+func (inst *Instance) ResolveReference(ctx context.Context, ref *dsref.Ref, mode string) (string, error) {
 	if inst == nil {
-		return "", dsref.ErrNotFound
+		return "", dsref.ErrRefNotFound
 	}
 
 	// Handle the "me" convenience shortcut
 	if ref.Username == "me" {
 		ref.Username = inst.cfg.Profile.Peername
 	}
 
-	resolvers, err := inst.resolveSources(source)
+	resolver, err := inst.resolverMode(mode)
```

Comment:
This reads really strangely. `resolverMode` follows the "noun adjective" pattern, as if it's a getter that retrieves the mode of a resolver. But it actually does the opposite, it takes a mode and returns a resolver. Maybe `resolverByMode` would work?